crates/polars-io/src/csv/buffer.rs

```diff
@@ -31,29 +35,115 @@ impl PrimitiveParser for Float64Type {
     }
 }
 
-impl PrimitiveParser for UInt32Type {
-    #[inline]
-    fn parse(bytes: &[u8]) -> Option<u32> {
-        lexical::parse(bytes).ok()
-    }
-}
-impl PrimitiveParser for UInt64Type {
-    #[inline]
-    fn parse(bytes: &[u8]) -> Option<u64> {
-        lexical::parse(bytes).ok()
-    }
-}
-impl PrimitiveParser for Int32Type {
-    #[inline]
-    fn parse(bytes: &[u8]) -> Option<i32> {
-        lexical::parse(bytes).ok()
-    }
+macro_rules! impl_integer_parser {
```

Comment:
Could we make this work with generics and then a macro that dispatches to the generics?